### DISCUSSION TITLE
6.4 makeDecorator: storyFn requires 2 parametors?

### DISCUSSION THREAD

**Discussion by NickMcBurney** [storybookjs/storybook#16875] [Help] [answered]
I have a decorator using syntax as described here: https://storybook.js.org/docs/vue/addons/addons-api#makedecorator-api

But with Storybook 6.4 I'm getting build errors because `storyFn` is expecting 2 arguments:

```
wrapper: (storyFn: StoryFn, context: StoryContext) => {
  /* ... some code ... */

  return storyFn(context); // this line errors 'Expected 2 arguments, but got 1.ts(2554)'
}
```

![image](https://user-images.githubusercontent.com/6604187/144450636-c7edf3d4-3dc5-46eb-aa3f-deda28028afe.png)

I can't find any guidance on this in docs, and the TS definitions aren't very helpful.

Does anyone know what `storyFn` is expecting as 2nd argument?

**Comment by shilman** [comment, marked as answer]
@NickMcBurney what version are you upgrading from? starting in 6.0, story functions take an `args` object as their first input, and the context object as their second https://storybook.js.org/blog/introducing-storybook-args/

**Reply by NickMcBurney** [reply]
I'm upgrading from 6.3.12 but haven't had this error before (and have only ever used single param)

**Reply by shilman** [reply]
Yes, we updated the types in 6.4 and it's possible that this is a bug, or more likely a bugfix that is also an unintentional breaking change cc @tmeasday

**Reply by tmeasday** [reply]
The story function in a decorator is a `LegacyStoryFn` and as such always takes one argument @shilman. This has not changed. @NickMcBurney can you post a reproduction?

**Reply by NickMcBurney** [reply]
@tmeasday Here is a simple reproduction of the issue: https://codesandbox.io/s/optimistic-meitner-95vv3?file=/src/customDecorator.ts

See line 11 of /src/customDecorator.ts

**Reply by tmeasday** [reply]
Hey @NickMcBurney sorry the repro isn't working properly: 
<img width="1049" alt="image" src="https://user-images.githubusercontent.com/132554/144793300-e346dece-9c5f-43ad-a01b-f70724773e5a.png">

It looks like maybe the build step isn't running properly in Code Sandbox. I'm not quite sure how to make it work. Any chance you could convert it to a git URL for me to clone (or provide further instructions)?

**Reply by NickMcBurney** [reply]
Hi @tmeasday, sorry for that, tbh I could see the issue without build so thought that was OK. Here is GIT repo to my branch with the update:
https://github.com/NickMcBurney/storybook-vue3-router/tree/storybook-6.4.x

**Reply by tmeasday** [reply]
Ahhh, OK. There was a type problem in your original code, this line:

```ts
    wrapper: (storyFn: StoryFn, context: StoryContext) => {
```

Should be

```ts
    wrapper: (storyFn: LegacyStoryFn, context: StoryContext) => {
```

But it didn't complain in 6.3 as the type of `ArgStoryFn` (the other "half" of `StoryFn`) was overly permissive.

**Reply by NickMcBurney** [reply]
@tmeasday thanks that sorted it now.

**Reply by tmeasday** [reply]
@NickMcBurney just to check, you didn't copy that line from anywhere in the docs that we should fix? (Fair enough if you don't recall :) )

**Reply by NickMcBurney** [reply]
@tmeasday I'm pretty sure I cobbled this together myself and didn't copy from docs. Think its entirely my own fault 🤣
